lib/Models/UserDrawing.ts

```diff
@@ -0,0 +1,438 @@
+import { autorun, computed, runInAction, reaction } from "mobx";
+import Cartesian3 from "terriajs-cesium/Source/Core/Cartesian3";
+import Color from "terriajs-cesium/Source/Core/Color";
+import createGuid from "terriajs-cesium/Source/Core/createGuid";
+import defaultValue from "terriajs-cesium/Source/Core/defaultValue";
+import PolygonHierarchy from "terriajs-cesium/Source/Core/PolygonHierarchy";
+import CallbackProperty from "terriajs-cesium/Source/DataSources/CallbackProperty";
+import CustomDataSource from "terriajs-cesium/Source/DataSources/CustomDataSource";
+import DataSource from "terriajs-cesium/Source/DataSources/DataSource";
+import Entity from "terriajs-cesium/Source/DataSources/Entity";
+import PolylineGlowMaterialProperty from "terriajs-cesium/Source/DataSources/PolylineGlowMaterialProperty";
+import isDefined from "../Core/isDefined";
+import DragPoints from "../Map/DragPoints";
+import ModelTraits from "../Traits/ModelTraits";
+import CreateModel from "./CreateModel";
+import MapInteractionMode from "./MapInteractionMode";
+import Terria from "./Terria";
+
+interface DragPoints {
+  updateDraggableObjects: (dataSource: DataSource) => void;
+  setUp: () => void;
+  getDragCount: () => number;
+  resetDragCount: () => void;
+}
+
+interface Options {
+  terria: Terria;
+  messageHeader?: string;
+  allowPolygon?: boolean;
+  onMakeDialogMessage?: () => string;
+  onPointClicked?: (dataSource: DataSource) => void;
+  onPointMoved?: (dataSource: DataSource) => void;
+  onCleanUp?: () => void;
+}
+
+class EmptyTraits extends ModelTraits {
+  static traits = {};
+}
+
+export default class UserDrawing extends CreateModel(EmptyTraits) {
```

Comment:
Why extend Model without any traits? I think it'd at least be useful to implement Mappable. Maybe because this shouldn't be shown in the workbench?

Maybe this should be a map overlay instead of a model? Currently there's no distinction, but maybe that'd be useful? I think we should talk about this.

Reply:
It has to be a `BaseModel` because `terria.overlays` is a `Workbench` and it references some non-Trait things like `uniqueId`. I made the traits empty because I couldn't think of any useful traits for overlays.

cc @kring

Reply:
Interesting. I didn't catch that there already was an overlays list